### PR TITLE
Adminsay color is now choices instead of color pick

### LINKED ISSUE #10159: [New Feature]: Admin chat should use "cfc (chat format color)" class
### Describe the feature request

Admin say chats allows custom color, but this is a bad design
* Admin using a white color chat makes their chat is not readable in white theme
* vice versa. Dark ones are not readable in dark theme
* Image: `dark-chat-theme.scss`

image

### PR DESCRIPTION
**Pull request by EvilDragonfiend**
## About The Pull Request
Adminsay color is now choices instead of color pick
* closes #10159
* a player chat color pick is bad. chat color should be standised.

## Why It's Good For The Game
bad chat readability bad

## Testing Photographs and Procedure

image

## Changelog
:cl:
code: Adminsay color is now choices instead of color pick
code: added cfc color defines.
/:cl:

### CODE REVIEW

File: tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/game_preferences/admin.tsx [hunk 1]
```diff
@@ -1,11 +1,34 @@
...
+      cfc_bluesky: 'Bluesky',
+      cfc_darkbluesky: 'Dark Bluesky',
+      cfc_navy: 'Navy',
+      cfc_blue: 'Blue',
+      cfc_indigo: 'Indigo',
+      cfc_purple: 'Purple',
+      cfc_violet: 'Violet',
+      cfc_magenta: 'Magenta',
+      cfc_redpurple: 'Redpurple', // pain. manual addition is innevitable
+    },
```
Comment (EvilDragonfiend) [inline tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/game_preferences/admin.tsx:26]: OOC chats should have this too, but I have no idea how I can make a module of this. I hate this.

**Comment by Tyranicranger4** [comment]
No fun allowed.

**Comment by Fronsis** [comment]
Uhh why are you closing the kilo map pr? Typo i guess

**Comment by EvilDragonfiend** [comment]
> Uhh why are you closing the kilo map pr? Typo i guess
> 
> #10159 this one i guess

oopsie accidentally wrote a wrong one. thaks

**Comment by PowerfulBacon** [comment]
Theres probably better ways to do this, like allowing any colour but inverting it if the contrast against the background is too low.

**Comment by EvilDragonfiend** [comment]
I am not certain if there's any way to adjust your color value dynamically.

**Comment by Rukofamicom** [comment]
Add an accessibility option to override custom chat colors instead, let the people who like custom colors have and use them.

**Review by Rukofamicom — CHANGES_REQUESTED** [review]
🔑 

If you're struggling to see colors because of your dark/light mode differences, make an accessibility option rather than taking away customization from others. Add an option that overrides custom colors with a color of your choice.

**Comment by EvilDragonfiend** [comment]
> 🔑
> 
> If you're struggling to see colors because of your dark/light mode differences, make an accessibility option rather than taking away customization from others. Add an option that overrides custom colors with a color of your choice.

That defeats the point of asay color.

**Comment by Rukofamicom** [comment]
It's better than forcing someone else to use a color they don't like as much. 
If you don't like the color they picked you should be changing it only for yourself.

**Comment by EvilDragonfiend** [comment]
> > That defeats the point of asay color.
> 
> Not any moreso than this PR does right now. If you have a problem with someone else's color it should only be changed for you rather than forcibly restricting others' choices.
> 
> We like the colors we have picked.

Yes, I know people like to pick the colors they like, but what you suggested with the key totally defeats the point of that.
The current design is to show "your color" to people. I am really not against this. The bad design here is that the color value is applied 100% to the system no matter how other people's background setting. The ideal solution is adjusting the value automatically by which background theme you're using, but I don't think that's possible.

and what you suggested is setting "a color of a chat" overriding which color you have. It's more destructive change. It's a different topic. If that option exists, what's the point of asay color picking?

another solution is to have 2 color settings for asay, but it's also needed thing to ooc chat (and perhaps more). This means doubled option for a single preference which requires a user to care how those look like in both of background themes.


Well, back to the main, I'd consider a way to adjust values dynamically based on a user's currently used background, but I really don't think that's possible for how the code works. It's the issue of the system design, not the user's behavior.
asay is a better place to start the approach rather than ooc, but actually ooc is where a solution is highly required. Imagine someone uses black font color. They might think their words are well visible if they think people only use light theme, but it's extremely disrupting to dark theme users.


By writing this by time, I realised that I didn't specify where the issue comes exactly.
The issue doesn't come from how user picks a color. It comes the game uses two themes, and that makes user's custom color an issue. `<font color='color_thing'>` isn't really compatible well with how TGUI chat works. That's why I implemented cfc colors with BB PR at first.

and for now, this is the solution as far as I can think and suggest.

**Comment by PowerfulBacon** [comment]
This is admin only and I don't think any of the admins will want this. If you want to implement a seperation between light mode and dark mode asay colour then feel free to re-open.

**Pull request closed without merging**